### PR TITLE
Throw exception when missing IWorkspaceTaskScheduleFactory implementation

### PR DESCRIPTION
**Pull request by robinsedlaczek**
Suggestion for fix for #6498.

### CODE REVIEW

**Comment by davkean** [comment]
Sorry for the late reply, we missed this.

tag @dotnet/roslyn-ide @mattwar 

:+1:

File: src/Workspaces/Core/Portable/Workspace/Workspace.cs [hunk 1]
```diff
@@ -60,6 +60,12 @@ protected Workspace(HostServices host, string workspaceKind)
 
             // queue used for sending events
             var workspaceTaskSchedulerFactory = _services.GetService<IWorkspaceTaskSchedulerFactory>();
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Workspace/Workspace.cs]: You can just call GetRequiredService instead that will throw for you if it doesn't exist.

**Comment by jasonmalinowski** [comment]
I don't disagree with intent, but there's a helper to already throw for you. If the text of that error is abysmal, feel free to update your well-written text into that helper instead.

**Comment by robinsedlaczek** [comment]
@jasonmalinowski @davkean Thanks for the reply! And the text of the exception is not abysmal. Nevertheless, I changed it. :) Just to show the name of the service type that is missing. Further, I use the GetRequiredService method in the Workspace ctor instead of throwing the exception there. Thanks for giving the hint for the helper method.

**Comment by jasonmalinowski** [comment]
:+1: If your git-fu is good you can elide the two commits of your initial change and git revert of that.

**Comment by davkean** [comment]
@robinsedlaczek If you could squash the first two commits, we can merge this in.

**Comment by robinsedlaczek** [comment]
I have some problems while squashing the first two commits. In order to save time, I like to close this PR and provide a clean new one.

**Pull request closed without merging**